### ISSUE TITLE
Hold Kernels not matched by install.sh regex

### ISSUE THREAD

**Issue by doertedev** [docker/docker#19977]
As a user suffering from "cloud"(tm) I might have to specifically mark my kernel as "hold" in order to not have my package manager betraying me. Ubuntu/Aufs problems lead me to apt-mark hold "linux.\*-$version" the specific version. Your `dpkg -l` regex matches ^ii.*, which won't work with "held" packages. They appear as "hi" instead.

```
hi  linux-headers-$major-$minor          $major-$minor-$patch                       all          Header files related to Linux kernel version $major
hi  linux-headers-$major-$minor-generic  $major-$minor-$patch                       amd64        Linux kernel headers for version $major on 64 bit x86 SMP
```

The corresponding regex is [here](https://github.com/docker/docker/blob/master/hack/install.sh#L375).

Greetings and thanks in advance.

**Comment by thaJeztah** [comment]
Thanks for reporting; would you be interested in opening a pull-request to adjust the regex?

**Comment by thaJeztah** [comment]
Also note that our installation guide explains the manual install procedure; https://docs.docker.com/engine/installation/ubuntulinux/. (The automated install script is only on a best-effort, because we cannot guarantee it to work on all possible combinations.

**Comment by calavera** [comment]
Fixed by #20038.
